clients/socks5/src/commands/upgrade.rs

```diff
@@ -3,16 +3,21 @@
 
 use crate::client::config::{Config, MISSING_VALUE};
 use clap::{App, Arg, ArgMatches};
-use client_core::config::default_validator_rest_endpoints;
-use config::defaults::DEFAULT_MIXNET_CONTRACT_ADDRESS;
+use config::defaults::default_api_endpoints;
 use config::NymConfig;
 use std::fmt::Display;
 use std::process;
-use version_checker::{parse_version, Version};
+use version_checker::Version;
+
+#[allow(dead_code)]
+fn fail_upgrade<D1: Display, D2: Display>(from_version: D1, to_version: D2) -> ! {
+    print_failed_upgrade(from_version, to_version);
+    process::exit(1)
+}
 
 fn print_start_upgrade<D1: Display, D2: Display>(from: D1, to: D2) {
     println!(
-        "\n==================\nTrying to upgrade client from {} to {} ...",
+        "\n==================\nTrying to client gateway from {} to {} ...",
```

Comment:
Still not good, but third time's a charm I guess :D

Reply:
How the hell did I manage to do that ... : D